node/actors/bft/src/lib.rs

```diff
@@ -96,34 +90,27 @@ impl Config {
             // This is the infinite loop where the consensus actually runs. The validator waits for either
             // a message from the network or for a timeout, and processes each accordingly.
             loop {
-                let deadline = *deadline_recv.borrow();
-                let input = pipe.recv.recv(&ctx.with_deadline(deadline)).await.ok();
+                let input = pipe.recv.recv(ctx).await;
 
                 // We check if the context is active before processing the input. If the context is not active,
                 // we stop.
                 if !ctx.is_active() {
                     return Ok(());
                 }
 
-                // Check if an asynchronous error was received.
```

Comment:
The error sender is within a loop within a background task, so regular error propagation isn't feasible AFAIU. Also when received. 
However, as in the previous implementation, errors are not returned, only reported locally in tracing/metrics.
If we are to keep this, meaning to have no errors which would lead to termination, then the error channel and handling of non-cancel, internal errors is indeed not needed. 
Is that the case?

Reply:
if the background task fails with an error, then the whole scope gets cancelled and also fails with an error. Background tasks and regular tasks behave the same in that aspect

Reply:
so, if the error is critical then the background task can just fail, and if it is not then it can log the error directly.

Reply:
I see, so since we don't need non-cancel error propagation support, this would be good enough. 

I removed the err channel and the bidirectional communication, changes here: https://github.com/matter-labs/era-consensus/pull/56/commits/2209d5dc32e7c6ce4195c19101faf4290b68535c